### PR TITLE
ecp: Unify test cases

### LINKED ISSUE #7306: Bignum - Optimise test methods for secp521r1,secp224r1, secp256r1 and secp384r1
### Suggested enhancement

The test functions between those eliptic curves, are very similar to each other e.g. [ecp_mod_p521_raw](https://github.com/Mbed-TLS/mbedtls/blob/development/tests/suites/test_suite_ecp.function#L1392-L1432)  and [ecp_mod_p224_raw](https://github.com/Mbed-TLS/mbedtls/blob/development/tests/suites/test_suite_ecp.function#L1349-L1389).

The ask is to unifiy that is to unify this testing function, using a generic template

~~~~~~
void ecp_mod_px_raw(char *input_N,
                    char *input_X,
                    int curve_id,
                    char *result)
~~~~~~

Slight differences in invoking those tests methods, such as the bitlength, limb sizes for different architectures, and the function pointer to be invoked can be mapped to the curve_id

### Justification

This will reduce the test code size.

### PR DESCRIPTION
**Pull request by minosgalanakis**
## Description

Resolves #7306 .

This patch replaces the test functions for the following curves:

- MBEDTLS_ECP_DP_SECP192R1
- MBEDTLS_ECP_DP_SECP224R1
- MBEDTLS_ECP_DP_BP256R1
- MBEDTLS_ECP_DP_BP512R1
- MBEDTLS_ECP_DP_BP384R1

Instead of invoking a functionally similar curve specific method, 
`ecp_mod_pXXX_raw(char *input_N, char *input_X, char *result)` ,
                          
the new generic implementation is accepting a curve_id, 
`
ecp_mod_p_generic_raw(int curve_id, char *input_N, char *input_X, char *result)` 

which is defined in the [ecp interface](https://github.com/Mbed-TLS/mbedtls/blob/development/include/mbedtls/ecp.h#L66-L76).

Then the curve specific parameters are switch-gated, greatly simplifying the test code.

## Gatekeeper checklist

- [x] **changelog** Not required since that is a new interface, and are not changing behaviour or logs.
- [x] **backport** Not required since that is a new interface not ported to 2.28
- [x] **tests** Provided

### CODE REVIEW

**Comment by minosgalanakis** [comment]
Please only review [this commit](https://github.com/Mbed-TLS/mbedtls/pull/7329/commits/6b83341b027500bcd1ab81d2a11b18cac9d7609b), since the pr is based on top of [PR7222](https://github.com/Mbed-TLS/mbedtls/pull/7222)

**Review by gabor-mezei-arm — DISMISSED** [review]
LGTM

**Comment by minosgalanakis** [comment]
I have added an extra [commit ](https://github.com/Mbed-TLS/mbedtls/pull/7329/commits/5ce38d326c3a819589a39393366d4266b75f5162)to avoid a regression of the issues fixed by #7103 and #7343.

**Review by gabor-mezei-arm — CHANGES_REQUESTED** [review]

**Comment by minosgalanakis** [comment]
@gabor-mezei-arm I agree that was not straightforward. I have updated the code during the rebase for the conflict resolution, and now dependenices can be optionally declared as class members. 

`arch_split` will still append the architectural dependency when set.

**Review by gabor-mezei-arm — APPROVED** [review]
LGTM

**Review by yanesca — CHANGES_REQUESTED** [review]
I have a minor suggestion for improvement and left a question in the comments.

File: tests/suites/test_suite_ecp.function [hunk 1]
```diff
@@ -1266,7 +1275,7 @@ exit:
 }
 /* END_CASE */
 
-/* BEGIN_CASE depends_on:MBEDTLS_TEST_HOOKS */
+/* BEGIN_CASE depends_on:MBEDTLS_TEST_HOOKS:MBEDTLS_ECP_DP_SECP_GENERIC_ENABLED */
```
[the remark this answers is not in the document]
Comment (minosgalanakis) [reply tests/suites/test_suite_ecp.function:1278]: As a matter of fact we don't. Good catch. Will be removed.

**Pull request merged**